utils/rest/client.cc

```
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2025 ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include <seastar/net/tls.hh>
+#include <seastar/net/dns.hh>
+#include <seastar/util/short_streams.hh>
+
+#include "client.hh"
+
+using namespace seastar;
+
+static std::string default_server_name(const std::string& host) {
+    // don't verify host cert name if "host" is just an ip address.
+    // typically testing.
+    bool is_numeric_host = seastar::net::inet_address::parse_numerical(host).has_value();
+    return is_numeric_host ? std::string{} : host;
+}
```

I don't think we even support TLS hostname validation for numeric hosts. The host is provided through the `tls_options::server_name` which serves two purposes:

1. Define the hostname for TLS SNI (so that a multi-domain server knows which certificate to serve).
2. Define the name for TLS hostname validation.

But since SNI only accepts DNS names, there is currently no way we can validate a certificate from a numeric host, unless the server silently ignores invalid SNI values (normally, I would expect it to abort the TLS handshake). Maybe we should be checking this earlier by not [ignoring the return value](https://github.com/scylladb/seastar/blob/26badcb14c1b8a03547d5b7fb24ab6b5f4c42299/src/net/tls.cc#L1225) of `gnutls_server_name_set()`.